features/authentication.feature

```diff
@@ -117,6 +117,24 @@ Feature: OAuth authentication
     Then the output should not contain "github.com password for mislav"
     And the file "../home/.config/hub" should contain "oauth_token: OTOKEN"
 
+  Scenario: Credentials from GITHUB_TOKEN
+    Given the GitHub API server:
+      """
+      get('/user') {
+        json :login => 'mislav'
+      }
+      post('/user/repos') {
+        assert({ "HTTP_AUTHORIZATION" => "token feline0123" }, env)
```

Comment:
I don't think you need to tweak the `assert` method here. How about simply using the approach like other tests?

```rb
halt 401 unless request.env['HTTP_AUTHORIZATION'] == 'token OTOKEN'
```